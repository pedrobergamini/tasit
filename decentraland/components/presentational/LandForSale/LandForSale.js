import React from "react";
import { StyleSheet, View } from "react-native";
import PropTypes from "prop-types";
import { responsiveWidth } from "react-native-responsive-dimensions";
import Estate from "@presentational/Estate";
import Parcel from "@presentational/Parcel";
import LandForSaleInfo from "@presentational/LandForSaleInfo";
import AssetTypes from "@constants/AssetTypes";

const { ESTATE, PARCEL } = AssetTypes;

export default function LandForSale({ landForSale }) {
  const { asset } = landForSale;
  const { type } = asset;

  return (
    <View style={styles.landContainer}>
      {(() => {
        switch (type) {
          case ESTATE:
            return <Estate estate={asset} />;
          case PARCEL:
            return <Parcel parcel={asset} />;
        }
      })()}
      <LandForSaleInfo landForSale={landForSale} />
    </View>
  );
}

LandForSale.propTypes = {
  landForSale: PropTypes.object.isRequired,
};

const styles = StyleSheet.create({
  landContainer: {
    width: responsiveWidth(95),
  },
});
