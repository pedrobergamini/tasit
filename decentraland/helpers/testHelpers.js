import AssetTypes from "@constants/AssetTypes";
const { ESTATE, PARCEL } = AssetTypes;

export const parcel = {
  type: PARCEL,
  id: "0123456789",
  name: "Sample Parcel",
  img: "https://api.decentraland.org/v1/parcels/0/0/map.png",
};

export const estate = {
  type: ESTATE,
  id: "123",
  name: "Sample Estate",
  img: "https://api.decentraland.org/v1/estates/1/map.png",
};

export const estateForSale = {
  id: "1",
  priceMana: 0,
  priceUSD: 0,
  asset: estate,
};

export const parcelForSale = {
  id: "2",
  priceMana: 0,
  priceUSD: 0,
  asset: parcel,
};

export const anAction = {
  getId: async () =>
    "0x1234567890123456789012345678901234567890123456789012345678901234",
};

export default {
  estateForSale,
  parcelForSale,
  parcel,
  estate,
  anAction,
};
