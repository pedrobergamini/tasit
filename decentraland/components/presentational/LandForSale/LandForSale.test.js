import React from "react";
import { shallow } from "enzyme";
import LandForSale from "@presentational/LandForSale";
import { estateForSale } from "@helpers/testHelpers";

describe("LandForSale", () => {
  it("renders the component", async () => {
    expect(
      shallow(<LandForSale landForSale={estateForSale} />)
    ).toMatchSnapshot();
  });
});
