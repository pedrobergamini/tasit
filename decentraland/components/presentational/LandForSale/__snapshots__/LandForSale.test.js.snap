// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandForSale renders the component 1`] = `
<Component
  style={
    Object {
      "width": 712.5,
    }
  }
>
  <Estate
    estate={
      Object {
        "id": "123",
        "img": "https://api.decentraland.org/v1/estates/1/map.png",
        "name": "Sample Estate",
        "type": "ESTATE",
      }
    }
  />
  <LandForSaleInfo
    landForSale={
      Object {
        "asset": Object {
          "id": "123",
          "img": "https://api.decentraland.org/v1/estates/1/map.png",
          "name": "Sample Estate",
          "type": "ESTATE",
        },
        "id": "1",
        "priceMana": "0",
        "priceUSD": 0,
      }
    }
  />
</Component>
`;
